homeassistant/components/weather/openweathermap.py

```diff
@@ -130,6 +130,9 @@ def humidity(self):
     @property
     def wind_speed(self):
         """Return the wind speed."""
+        if self.hass.config.units.name == 'imperial':
+            return round(self.data.get_wind().get('speed') * 2.24, 2)
```

Comment:
I think that we should implement this on a weather component level, just like the climate component will look at the unit and convert current temp, target temp etc.

Reply:
Although that might mean we have to expose a unit per property, as weather reports per country vary in units.

Reply:
Agreed.